src/app/container/view/view.component.ts

```diff
@@ -82,7 +82,7 @@ export class ViewContainerComponent extends View implements OnInit {
   ngOnInit() {
     this.isPublic$ = this.sessionQuery.isPublic$;
     this.toolQuery.tool$.pipe(takeUntil(this.ngUnsubscribe)).subscribe(tool => {
-      this.tool = tool;
+      this.tool = JSON.parse(JSON.stringify(tool));
```

Comment:
:open_mouth:

Reply:
I see Andrew already commented, but what is the purpose of this?

Reply:
Akita and probably most state management tools makes everything immutable.  However, we need to allow the UI to make modifications (edit).  Spread operator and Object.assign both have problems either because it doesn't copy enough properties (no deep copy) or it does deep copy and copies the read-only attribute too.